Content.Server/PDA/PDASystem.cs

```diff
@@ -106,7 +109,59 @@ private void UpdatePDAUserInterface(PDAComponent pda)
 
             var address = GetDeviceNetAddress(pda.Owner);
             var hasInstrument = HasComp<InstrumentComponent>(pda.Owner);
-            var state = new PDAUpdateState(pda.FlashlightOn, pda.PenSlot.HasItem, ownerInfo, pda.StationName, false, hasInstrument, address);
+            TimeSpan currentTime;
+            currentTime = GameTicker.RoundDuration();
+
+            var stationTime = new StationTimeText
+            {
+                Hours = currentTime.Hours.ToString(),
+                Minutes = currentTime.Minutes.ToString(),
+            };
+
+            var stationAlert = new StationAlert
+            {
+                Level = null,
+                Color = Color.White
+            };
+
+            var stationUid = _stationSystem.GetOwningStation(pda.Owner);
+            if (TryComp(stationUid, out AlertLevelComponent? alertComp) &&
+                alertComp.AlertLevels != null)
+            {
+                stationAlert.Level = alertComp.CurrentLevel;
+                if (alertComp.AlertLevels.Levels.TryGetValue(alertComp.CurrentLevel, out var details))
+                    stationAlert.Color = details.Color;
+            }
+
+            List<string> accessLevels;
+            if (pda.IdSlot.Item is { Valid: true } targetId)
+            {
+                var accessLevels_enum = _access.TryGetTags(targetId) ?? new List<string>();
+                accessLevels = accessLevels_enum.ToList();
+            }
+            else
+            {
+                accessLevels = new List<string>();
+            }
+
+            var accessLevelsConvert = new List<string>();
+            accessLevels.RemoveAll(s => s.Contains("EmergencyShuttleRepealAll"));
+            foreach (var access in accessLevels)
+            {
+                if (!_prototypeManager.TryIndex<AccessLevelPrototype>(access, out var accessLevel))
+                {
+                    Logger.ErrorS(SharedIdCardConsoleSystem.Sawmill, $"Unable to find accesslevel for {access}");
+                    continue;
+                }
+
+                accessLevelsConvert.Add(GetAccessLevelName(accessLevel));
+            }
```

Comment:
I don't think PDA should be revealing access levels as this can be used to quickly powergame checking for AA.

Reply:
Do you mean the microwave feature or the fact that the security service can look at the PDA and see that the player has access that should not be?

Reply:
> Do you mean the microwave feature or the fact that the security service can look at the PDA and see that the player has access that should not be?

the latter